src/dbSta/src/dbNetwork.cc

```diff
@@ -3035,14 +3037,44 @@ void dbNetwork::hierarchicalConnect(dbITerm* source_pin,
   // in hierarchy, which is ok, and the source/dest modnet will be null
   dbModNet* source_db_mod_net = source_pin->getModNet();
   dbModNet* dest_db_mod_net = dest_pin->getModNet();
-  // case 1: source/dest in same module
+
+  //
+  // make sure there is a direct flat net connection
+  // Recall the hierarchical connections are overlayed
+  // onto the flat db network, so we have both worlds
+  // co-existing, something we respect even when making
+  // new hierarchical connections.
+
+  dbNet* source_db_net = source_pin->getNet();
+  dbNet* dest_db_net = dest_pin->getNet();
+
+  if (!source_db_net) {
+    std::string connection_name_str(connection_name);
+    std::string flat_name = connection_name_str + "_flat";
+    source_db_net = dbNet::create(block(), flat_name.c_str(), false);
+    source_pin->connect(source_db_net);
+  }
+
+  if (dest_db_net && (dest_db_net != source_db_net)) {
+    // if we have dest_mod_net, keep it
```

Comment:
What does it mean to connect two pins belonging to different dbNets?  Is there an implied transfer?  If so, how can you keep the dest_mod_net?

Reply:
I think it best we talk this through as it is hard to explain in text form. What this section of code is doing is making sure there is a flat net as well as a hierarchical net going from source_pin to dest_pin. Now it is possible that this code is called in some intermediate state (as seems to be the case in cts) so it disconnects the dest_pin, connects it to the source pin AND if the dest_pin was originally connected to a modnet (a boundary case) preserves that modnet connection. That code addresses something found when running gcd through the whole hierarchical flow and can be removed if required from this pull request (it won't break any of the regressions).

Reply:
As per our discussion, I have removed this block of code.